qiskit/opflow/primitive_ops/pauli_sum_op.py

```diff
@@ -251,7 +251,8 @@ def to_matrix(self, massive: bool = False) -> np.ndarray:
 
     def __str__(self) -> str:
         def format_sign(x):
-            return x.real if np.isreal(x) else x
+            threshold = 1e-10
```

Comment:
I'm wondering if instead of using a hardcoded `1e-10` here we should defer to the default `atol` which I think is `1e-8` but is also configurable. Something like:

```suggestion
            threshold = self.primitive.atol
```